### ISSUE TITLE
Simple Keras training

### ISSUE THREAD

**Issue by mortendahl** [tf-encrypted/tf-encrypted#540]
Private training of simple model, eg logistic regression, using the Keras API. This involves adding backwards logic for involved layers.

### Next Steps

- [x] Add backward pass to:
  - [x] Dense Layer
  - [x] Activation functions
- [ ] Loss functions:
  - [x] Cross Entropy
  - [ ] MSE
- [x] [compile ](https://www.tensorflow.org/api_docs/python/tf/keras/models/Model#compile)
- [x] [fit](https://www.tensorflow.org/api_docs/python/tf/keras/models/Model#fit)
- [ ] [evaluate](https://www.tensorflow.org/api_docs/python/tf/keras/models/Model#evaluate)
